src/mono/wasm/build/WasmApp.targets

```diff
@@ -154,10 +154,23 @@
     <WasmStripILAfterAOT Condition="'$(WasmStripILAfterAOT)' == ''">false</WasmStripILAfterAOT>
 
     <WasmRuntimeAssetsLocation Condition="'$(WasmRuntimeAssetsLocation)' == ''">_framework</WasmRuntimeAssetsLocation>
-    <BlazorCacheBootResources Condition="'$(BlazorCacheBootResources)' == ''">true</BlazorCacheBootResources>
+
+    <!-- Default values for build properties -->
+    <InvariantTimezone Condition="'$(InvariantTimezone)' == ''">false</InvariantTimezone>
   </PropertyGroup>
 
   <ItemGroup>
+    <!-- Runtime feature defaults to trim unnecessary code -->
+    <RuntimeHostConfigurationOption Condition="'$(HybridGlobalization)' == ''" Include="System.Globalization.Hybrid" Value="false"/>
+    <RuntimeHostConfigurationOption Condition="'$(InvariantGlobalization)' == ''" Include="System.Globalization.Invariant" Value="false" />
+    <RuntimeHostConfigurationOption Condition="'$(EventSourceSupport)' == ''" Include="System.Diagnostics.Tracing.EventSource.IsSupported" Value="false" />
+    <RuntimeHostConfigurationOption Condition="'$(UseSystemResourceKeys)' == ''" Include="System.Resources.UseSystemResourceKeys" Value="true" />
+    <RuntimeHostConfigurationOption Condition="'$(EnableUnsafeUTF7Encoding)' == ''" Include="System.Text.Encoding.EnableUnsafeUTF7Encoding" Value="false" />
+    <RuntimeHostConfigurationOption Condition="'$(HttpActivityPropagationSupport)' == ''" Include="System.Net.Http.EnableActivityPropagation" Value="false" />
+    <RuntimeHostConfigurationOption Condition="'$(NullabilityInfoContextSupport)' == ''" Include="System.Reflection.NullabilityInfoContext.IsSupported" Value="false" />
+    <RuntimeHostConfigurationOption Condition="'$(_AggressiveAttributeTrimming)' == ''" Include="System.AggressiveAttributeTrimming" Value="true" />
+    <RuntimeHostConfigurationOption Condition="'$(DebuggerSupport)' == ''and '$(Configuration)' != 'Debug'" Include="System.Diagnostics.Debugger.IsSupported" Value="false" />
```

Comment:
I'm curious about this. I don't know when this happens during the build... but ideally this would happen pretty early on. The reasoning is that other parts of build/publish may look at the values here (ILLink, AOT compilers do, but there might be other parts as well).

The other potential problem is if somebody sets the item group directly. This is something we're not good at in other parts of the SDK, but it is a real scenario. Sometimes people add the value to the `RuntimeHostConfigurationOption` directly, mainly because it's a documented way how to set properties for runtime. We've had bugs in the tools where they got confused by getting the same value specified multiple times. We don't have a good solution how to handle that case, unless everything uses the properties, then it works pretty well.

So, I guess my main question is - why don't we manipulate the properties instead?

For example, for console app trimming, the property values are set to their trimming defaults very early on, so that other parts of the build can look at them and potentially modify the default value. The general recommendation is to use the property values to handle the feature switches everywhere possible (avoids the item group problem mentioned above).

Reply:
> "Why don't we manipulate the properties instead?"

I was trying to fix the trimming tests. It "fixed" them but I did not see that I reverted the size changes by that as well, so it is an incorrect change. Why do we have a problem here:
let's take e.g. `InvariantGlobalizationTrue` where we expect the non-invariant parts to be trimmed. We set the invariant mode using feature switch:
https://github.com/dotnet/runtime/blob/0f4196f67a1981dc414d9845629b13e00ce3f4e0/src/libraries/System.Runtime/tests/TrimmingTests/System.Runtime.TrimmingTests.proj#L24
and we expect it will work. However, the flow is:
1) evaluate project's build properties (we do not set `InvariantGlobalization` property, only the switch like this, so it's undefined after this step)
```
  <ItemGroup>
    <RuntimeHostConfigurationOption Include="System.Globalization.Invariant" Value="true" Trim="true" />
  </ItemGroup>
```
2) evaluate runtime build's properties from `WasmApp.targets` (it was undefined, so we set it to `false`, regardless of the feature's value)
3) linker time, the property for `Invariant` is set to false, so we do not trim non-invariant code
4) `InvariantGlobalizationTrue` checks if non-invariant code is there and it throws.

It's only the test's problem, on a sample app it works fine (users normally set properties, not features switches). In order to fix this situation we would could change trimming tests logic to set properties instead of feature switches. But I do not understand why we preferred to use switches over properties in the tests, so I was trying not to touch the current state. I will update the PR to show what I mean exactly.

Reply:
If this is an infra only change... then I don't mind at all (I understand it's complicated).
I'm mostly worried if this is changing how a WASM app is built for users.

Reply:
Yes, the differences are only for the setup that was leaving properties undefined.
The following switches are passed to linker with different values with this PR:
- `System.AggressiveAttributeTrimming` is true by default now, so it will be a breaking change,
- `System.Globalization.Invariant`, `System.Globalization.Hybrid`, `System.Reflection.NullabilityInfoContext.IsSupported` are false.
Without this PR they were not passed at all.

Reply:
Thanks for the explanation